xclim/indices/run_length.py

```diff
@@ -1166,41 +1166,49 @@ def suspicious_run_1d(
     Returns
     -------
     numpy.ndarray
-      Whether or not the series contains a run of identical values.
+      Whether or not the data points is part of a run of identical values.
     """
     v, rl, pos = rle_1d(arr)
+    sus_runs = rl >= window
     if thresh:
         if op in {">", "gt"}:
-            return ((v > thresh) * rl >= window).any()
-        if op in {"==", "eq"}:
-            return ((v == thresh) * rl >= window).any()
-        if op in {">=", "gteq"}:
-            return ((v >= thresh) * rl >= window).any()
-        raise NotImplementedError(f"{op}")
-    return (rl >= window).any()
+            sus_runs = sus_runs & (v > thresh)
+        elif op in {"==", "eq"}:
+            sus_runs = sus_runs & (v == thresh)
+        elif op in {">=", "gteq"}:
+            sus_runs = sus_runs & (v >= thresh)
+        else:
+            raise NotImplementedError(f"{op}")
+
+    out = np.zeros_like(arr, dtype=bool)
+    for st, l in zip(pos[sus_runs], rl[sus_runs]):
+        out[st : st + l] = True
+    return out
 
 
 def suspicious_run(
     arr: xr.DataArray,
     dim: str = "time",
     window: int = 10,
-    op: Optional[str] = None,
+    op: Optional[str] = ">",
```

Comment:
I'm wondering now if this is actually optional. Can we send in a `None`? If no, it's not `Optional`.